controllers/user.js

```diff
@@ -0,0 +1,239 @@
+const jwt = require("jsonwebtoken");
+const Joi = require("joi");
+const bCrypt = require("bcryptjs");
+const dotenv = require("dotenv");
+dotenv.config();
+const {
+    findUserByEmail,
+    updateKeyForUser,
+    findUserByToken,
+} = require("../service/user.service");
+
+const User = require("../models/user.model");
+const SECRET = process.env.SECRET;
+
+const userSchema = Joi.object({
+    email: Joi.string().email({ minDomainSegments: 2 }).required(),
+    password: Joi.string().min(7).required(),
+});
+
+const userSubscriptionSchema = Joi.object({
+    subscription: Joi.string().valid("starter", "pro", "business").required(),
+});
+
+const hashPassword = async (password) => {
+    const salt = await bCrypt.genSalt(10);
+    const hash = await bCrypt.hash(password, salt);
+    return hash;
+};
+
+const validatePassword = (password, hash) => bCrypt.compare(password, hash);
+
+const passwordValidator = async (password, userPassword) => {
+    const isValidPassword = await validatePassword(password, userPassword);
+    return isValidPassword;
+};
+
+const findUser = async (email) => {
+    try {
+        const user = await findUserByEmail(email);
+        return user;
+    } catch (err) {
+        console.error(err);
+    }
+};
+
+
+
+const signup = async (req, res, _) => {
+    try {
+        const {value, error} = userSchema.validate(req.body);
+        const {email, password} = value;
+
+        if(error) {
+            return res.status(400).json({ message: error.message });
+        }
+
+        const toLowerCaseEmail = email.toLowerCase();
```

Comment:
```suggestion
        const lowerCasedEmail = email.toLowerCase();
```
Tamto sugerowało działanie. A to stan.